Docs/mkdocs material (#447)
piskoviste/pisek: docs/task-parts/batch-checker.md

It should say whether the contestant output is correct.

There are various types of checkers you can use:

> - `tokens` - fast, versatile file equality checker
> - `diff` - file equality checker based on the `diff` command line tool (don't use, as it has a quadratic time complexity)
> - `judge` - custom checker

Reviewer: ```suggestion
- `tokens` – fast, versatile file equality checker
- `diff` – file equality checker based on the `diff` command line tool (avoid this option, as it has a quadratic time complexity)
- `judge` – custom checker
```

You can customize the tokens judge with `tokens_ignore_newlines` or `tokens_ignore_case`.
For comparing floats, set `tokens_float_rel_error` and `tokens_float_abs_error`.

> Details can be found in [config-documentation](./config-docs).

Reviewer: Broken link!
```suggestion
Details can be found in [config-documentation](../config-docs).
```

An equality checker based on the `diff` tool. Runs `diff -Bbq` under the hood.
Ignores whitespace and empty lines.

??? danger "This `out_check` is not recommended"

>     In some cases `diff` has quadratic complexity, leading to unexpectedly slow checking of outputs.

Reviewer: ```suggestion
    In some cases, `diff` has quadratic time complexity, leading to unexpectedly slow checking of outputs.
```

> and `judge_needs_in`, `judge_needs_out` to `0`/`1` depending whether the judge needs the input and the correct output.

Reviewer: ```suggestion
and `judge_needs_in`, `judge_needs_out` to `0`/`1`, depending on whether the judge needs the input and the correct output.
```

**Unlike the CMS documentation specifies, the files should be single-line only.**
There will be a warning otherwise.

??? example "Example `cms-batch` judge"

	For a [task](https://github.com/piskoviste/pisek/blob/master/examples/cms-batch) of printing *N* positive integers that sum up to *K*,

> 	the judge may look like this:
>     ```py
>     --8<-- "examples/cms-batch/gen.py"
>     ```

Reviewer: Wrong link!
```suggestion
	the judge may look like this:
    ```py
    --8<-- "examples/cms-batch/judge.py"
    ```
```

Optionally, the judge can write a one-line message for the contestant
to stderr (at most 255 bytes), followed by a sequence of lines with `KEY=value` pairs.
The following keys are allowed:

> - `POINTS` — Number of points awarded for this test case (used only if the exit code says "OK").
> - `LOG` — A message that should be logged.
> - `NOTE` — An internal note recorded in the database, but not visible to contestants.

Reviewer: An en dash is more appropriate here.
```suggestion
- `POINTS` – Number of points awarded for this test case (used only if the exit code says "OK").
- `LOG` – A message that should be logged.
- `NOTE` – An internal note recorded in the database, but not visible to contestants.
```

Values are again limited to 255 bytes.

### Opendata-v1 judge

> The opendata-v1 judge is the same as opendata-v2, with the exception of using different
> returncodes, returncode 0 for a correct output and returncode 1 for a wrong output.
> 
> However, usage of this judge type is discouraged as these returncodes can
> also be the result of an internal judge bug.
> 
> ??? danger "This `judge_type` is not recommended"
> 
>     Return with exitcode 1 is very common and is for example trigger by any exception in Python.
>     This 

Reviewer: ```suggestion
The opendata-v1 judge is the same as opendata-v2, with the exception of using different
return codes, return code 0 for a correct output and return code 1 for a wrong output.

However, usage of this judge type is discouraged as these return codes can
also be the result of an internal judge bug.

??? danger "This `judge_type` is not recommended"

    Returning with exit code 1 is very common and is for example triggered by any exception in Python.
```
